### PR TITLE
Canonical email block

### PR DESCRIPTION
**Pull request by JacksonSamuel42**

### CODE REVIEW

File: src/repositories/admin/canonical-email-block-repository.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  /** The hash to test against. If email is not provided, this parameter is required. */
+  readonly canonicalEmailHash?: never;
+}
+
+export interface CreateCanonicalEmailBlockParamsWithCanonicalEmailHash {
+  /** The hash to test against. If email is not provided, this parameter is required. */
+  readonly canonicalEmailHash: string;
+  /** The email to canonicalize, hash, and block. If this parameter is provided, canonical_email_hash will be ignored. */
+  readonly email?: string | null;
+}
```
Comment (neet) [inline src/repositories/admin/canonical-email-block-repository.ts:26]: The doc says 

- email: "If this parameter is provided, canonical_email_hash will be ignored."
- canonical_email_hash: "If email is not provided, this parameter is required."

Thus, if we provide

| # | Condition | |
| :-  | :-- | :-- |
| 1 |  Only `email` | `email` will be used |
| 2 | Only `canonical_email_hash` | `canonical_email_hash` will be used |
| 3 | Neither `email` nor `canonical_email_hash` | They say "If email is not provided, this parameter is required." so it is invalid. |
| 4 | Both `email` and `canonical_email_hash` | `canonical_email_hash` will be ignored, which will work the same as ＃1 |

This in turn, the proper definition should be

```ts
export interface CreateCanonicalEmailBlockParamsWithEmail {
  /** The email to canonicalize, hash, and block. If this parameter is provided, canonical_email_hash will be ignored. */
  readonly email: string;
}

export interface CreateCanonicalEmailBlockParamsWithCanonicalEmailHash {
  /** The hash to test against. If email is not provided, this parameter is required. */
  readonly canonicalEmailHash: string;
}
```

**Review by neet — CHANGES_REQUESTED** [review]

File: src/repositories/admin/canonical-email-block-repository.ts (hunk 1 above)
Comment (JacksonSamuel42) [reply src/repositories/admin/canonical-email-block-repository.ts]: yh you right

**Comment by JacksonSamuel42** [comment]
Hey neet, I'm creating masto snippets for vscode that I can add in the readme file for recommendation for those just starting to use masto. and of course if it's not a problem

https://marketplace.visualstudio.com/items?itemName=JacksonSamuel.masto-js-snippets

**Comment by neet** [comment]
I was planning to create a section that lists bots, apps, and related works using Masto.js on README.md. I'll promise to put yours as well when it's ready!

**Pull request merged**